src/doorman/Xania.cpp

```diff
@@ -0,0 +1,176 @@
+#include "Xania.hpp"
+
+#include "Channel.hpp"
+#include "Types.hpp"
+#include "doorman.hpp"
+
+#include <chrono>
+#include <cstdio>
+#include <cstdlib>
+#include <sys/socket.h>
+#include <sys/uio.h>
+#include <sys/un.h>
+#include <unistd.h>
+
+using namespace std::literals;
+
+static constexpr auto time_between_retries = 10s;
+
+namespace {
+void *iovec_cast(const void *thing) { return const_cast<void *>(thing); }
+}
+
+void Xania::SendToMUD(const Packet &p, const void *payload) {
+    if (!connected())
+        return;
+    const iovec vec[2] = {{iovec_cast(&p), sizeof(Packet)}, {iovec_cast(payload), p.nExtra}};
+    // TODO writev? rephrase this whole thing?
+    if (::writev(fd_.number(), vec, 2) != static_cast<ssize_t>(sizeof(Packet) + p.nExtra)) {
```

Comment:
Interesting use of writev, we didn't have that before.

Reply:
Right, the checking of return values for multiple write() calls was annoying me, so early on in the conversion (before I made it `throw`, something I'm still considering changing back), it's nicer to have one call to check :)

`writev` is great though for exactly this stuff. I think there's a good c++-ey solution to wrapping writev() nicely, hence the `TODO`

Reply:
(have updated the list of things to do in the PR description accordingly)